glfw.go

```diff
@@ -96,19 +113,28 @@ func (m *windowManager) glfwCursorPosCallback(window *glfw.Window, x, y float64)
 	m.sendPointerEvent(window, m.pointerPhase, x, y)
 }
 
-func (m *windowManager) glfwMouseButtonCallback(window *glfw.Window, key glfw.MouseButton, action glfw.Action, mods glfw.ModifierKey) {
-	if key == glfw.MouseButton1 {
-		x, y := window.GetCursorPos()
+func (m *windowManager) sendButtonEvent(window *glfw.Window, action glfw.Action, buttons embedder.PointerButtonMouse) {
+	if action == glfw.Press {
+		m.sendButtonPointerEvent(window, embedder.PointerPhaseDown, buttons)
+		m.pointerPhase = embedder.PointerPhaseMove
+	}
 
-		if action == glfw.Press {
-			m.sendPointerEvent(window, embedder.PointerPhaseDown, x, y)
-			m.pointerPhase = embedder.PointerPhaseMove
-		}
+	if action == glfw.Release {
+		m.sendButtonPointerEvent(window, embedder.PointerPhaseUp, buttons)
+		m.pointerPhase = embedder.PointerPhaseHover
+	}
+}
 
-		if action == glfw.Release {
-			m.sendPointerEvent(window, embedder.PointerPhaseUp, x, y)
-			m.pointerPhase = embedder.PointerPhaseHover
-		}
+func (m *windowManager) glfwMouseButtonCallback(window *glfw.Window, key glfw.MouseButton, action glfw.Action, mods glfw.ModifierKey) {
+	switch key {
+	case glfw.MouseButtonLeft:
+		m.sendButtonEvent(window, action, embedder.PointerButtonMousePrimary)
+	case glfw.MouseButtonRight:
+		m.sendButtonEvent(window, action, embedder.PointerButtonMouseSecondary)
+	case glfw.MouseButtonMiddle:
+		m.sendButtonEvent(window, action, embedder.PointerButtonMouseMiddle)
+	default:
+		m.sendButtonEvent(window, action, 1<<uint(key))
```

Comment:
How does this work? Does `1<<uint(key)` always map to a avlied `emberdder.PointerButton*` constant?

Reply:
Nope, [_If a mouse has more than five buttons, send higher bit shifted values_](https://github.com/flutter/engine/blob/8812781d912f85db71d64a2ec865782d3a2e92bf/shell/platform/embedder/embedder.h#L330-L331).

Reply:
`key`value is send from GLFW which are corresponding to [this](https://www.glfw.org/docs/latest/group__buttons.html).
Like I said in the at the beginning of the PR. 
`1<<uint(key)` suffice for all buttons.

Reply:
Awesome!

Reply:
I have a Logitech G700s.... It has 14 buttons.... :mouse: